File: Selenium/Activities/Activity8.java
package activity;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity8 {

	public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/dynamic-controls");
        driver.manage().window().maximize();
        System.out.println("Page Title: " + driver.getTitle());
        
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        
        By checkboxLocator = By.id("checkbox");
        By toggleButton = By.xpath("//button[text()='Toggle Checkbox']");

        driver.findElement(toggleButton).click();
        wait.until(ExpectedConditions.invisibilityOfElementLocated(checkboxLocator));
        System.out.println("Checkbox removed");
        driver.findElement(toggleButton).click();
        
        WebElement checkbox = wait.until(ExpectedConditions.presenceOfElementLocated(checkboxLocator));
        System.out.println("Checkbox appeared");

        checkbox.click();
        System.out.println("Checkbox selected");
        driver.quit();      
    }
}

File: Selenium/Activities/activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


driver = webdriver.Firefox()

try:
    
    driver.get("https://training-support.net/webelements/dynamic-controls")
    driver.maximize_window()

    
    print("Page Title:", driver.title)

    
    wait = WebDriverWait(driver, 10)

    
    checkbox_locator = (By.ID, "checkbox")
    toggle_button = (By.XPATH, "//button[text()='Toggle Checkbox']")

    
    driver.find_element(*toggle_button).click()

    
    wait.until(EC.invisibility_of_element_located(checkbox_locator))
    print("Checkbox removed")

    
    driver.find_element(*toggle_button).click()

    
    checkbox = wait.until(
        EC.presence_of_element_located(checkbox_locator)
    )
    print("Checkbox appeared")

    
    checkbox.click()
    print("Checkbox selected")

finally:
    
    driver.quit()